github_actions/lib/dependabot/github_actions/update_checker.rb

```diff
@@ -23,10 +23,20 @@ def latest_resolvable_version_with_no_unlock
         dependency.version
       end
 
-      def updated_requirements
-        return dependency.requirements if updated_source == dependency_source_details
+      def updated_requirements # rubocop:disable Metrics/PerceivedComplexity
+        previous = dependency_source_details
+        updated = updated_source
+        return dependency.requirements if updated == previous
+
+        if previous[:type] == "git" &&
```

Comment:
Thoughts on adding a comment explaining that this will maintain a short sha if latest?

Reply:
```suggestion
        # Maintain a short git hash only if it matches the latest
        if previous[:type] == "git" &&
```
SGTM!